File: 0780-reaching-points/0780-reaching-points.py
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        if sx>tx or sy>ty: return False;
        
        if tx == sx and (ty - sy)%tx == 0 : return True
        if ty == sy and (tx - sx)%ty == 0 : return True
        
        if tx>ty: return self.reachingPoints(sx, sy, tx%ty, ty)
        if tx<ty: return self.reachingPoints(sx, sy, tx, ty%tx)
        
        return False

File: 0780-reaching-points/0780-reaching-points.java
class Solution {
    public boolean reachingPoints(int sx, int sy, int tx, int ty) {
        if(sx>tx || sy>ty) return false;
        if(sx == tx && (ty - sy)%tx == 0) return true;
        if(sy == ty && (tx - sx)%ty == 0) return true;
        if(tx>ty) return reachingPoints(sx, sy, tx%ty, ty);        
        if(tx<ty) return reachingPoints(sx, sy, tx, ty%tx);
        return false;
    }
}